crates/core/tedge_mapper/src/aws/mapper.rs

```diff
@@ -35,25 +26,9 @@ impl TEdgeComponent for AwsMapper {
 
     async fn start(
         &self,
-        tedge_config: TEdgeConfig,
+        _tedge_config: TEdgeConfig,
         _config_dir: &Path,
     ) -> Result<(), anyhow::Error> {
-        let add_timestamp = tedge_config.query(AwsMapperTimestamp)?.is_set();
-        let mqtt_port = tedge_config.query(MqttClientPortSetting)?.into();
-        let mqtt_host = tedge_config.query(MqttClientHostSetting)?;
-        let clock = Box::new(WallClock);
-        // Quotas at: https://docs.aws.amazon.com/general/latest/gr/iot-core.html#limits_iot
-        let size_threshold = SizeThreshold(128 * 1024);
-
-        let converter = Box::new(AwsConverter::new(add_timestamp, clock, size_threshold));
-
-        let mut mapper = create_mapper(AWS_MAPPER_NAME, mqtt_host, mqtt_port, converter).await?;
-
-        mapper
-            .run(None)
-            .instrument(info_span!(AWS_MAPPER_NAME))
-            .await?;
-
-        Ok(())
+        panic!("This method should no more be used");
```

Comment:
I know that this if after my suggestion that you added this panic, however, with the new design, it would be by far nicer to call the  `run_mapper()` function you added in `main.rs`.

Even  better, as this function holds both Azure and AWS, to move here the AWS specific code:

```suggestion
            let (mut runtime, mut mqtt_actor) = start_basic_actors(mapper_name, &config).await?;

            let aws_converter = AwsConverter::new(
                config.query(AwsMapperTimestamp)?.is_set(),
                clock,
                SizeThreshold(255 * 1024),
            );
            let mut aws_converting_actor = ConvertingActor::builder("AwsConverter", aws_converter);
            mqtt_actor.register_peer(
                AwsConverter::in_topic_filter(),
                aws_converting_actor.get_sender(),
            );
            aws_converting_actor.register_peer(NoConfig, mqtt_actor.get_sender());

            runtime.spawn(aws_converting_actor).await?;
            runtime.spawn(mqtt_actor).await?;
            runtime.run_to_completion().await?;
```

Reply:
If we call `run_mapper`  in `start` then we have to move this `runtime` to the mapper.rs. IIRC the first implementation was the same, each mapper had their implementation and was starting the specific mapper actors.

Reply:
refactored